src/Features/Core/Portable/ExtractClass/ExtractClassWithDialogCodeAction.cs

```diff
@@ -0,0 +1,264 @@
+            }
+            else
+            {
+                // If user click cancel button, options will be null and hit this branch
+                return SpecializedCollections.EmptyEnumerable<CodeActionOperation>();
+            }
+        }
+
+        private async Task<Solution> PullMembersUpAsync(
+            Solution solution,
+            INamedTypeSymbol newType,
+            AnnotatedSymbolMapping symbolMapping,
+            ImmutableArray<ExtractClassMemberAnalysisResult> memberAnalysisResults,
+            CancellationToken cancellationToken)
+        {
+            using var _ = ArrayBuilder<(ISymbol member, bool makeAbstract)>.GetInstance(out var pullMembersBuilder);
+            using var _1 = ArrayBuilder<ExtractClassMemberAnalysisResult>.GetInstance(memberAnalysisResults.Length, out var remainingResults);
+            remainingResults.AddRange(memberAnalysisResults);
+
+            // For each document in the symbol mappings, we want to find the annotated nodes
+            // of the members and get the current symbol that represents those after
+            // any changes we made before members get pulled up into the base class.
+            // We only need to worry about symbols that are actually being moved, so we track
+            // the symbols from the member analysis. 
+            foreach (var documentId in symbolMapping.DocumentIds)
+            {
+                if (remainingResults.Count == 0)
+                {
+                    // All symbols have been taken care of
+                    break;
+                }
+
+                var document = solution.GetRequiredDocument(documentId);
+                var syntaxFacts = document.GetRequiredLanguageService<ISyntaxFactsService>();
+                var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                using var _2 = ArrayBuilder<ExtractClassMemberAnalysisResult>.GetInstance(remainingResults.Count, out var resultsToRemove);
+
+                // Out of the remaining members that we need to move, does this
+                // document contain the definition for that symbol? If so, add it to the builder
+                // and remove it from the symbols we're looking for
+                foreach (var memberAnalysis in remainingResults)
+                {
+                    var annotation = symbolMapping.SymbolToDeclarationAnnotationMap[memberAnalysis.Member];
+
+                    var nodeOrToken = root.GetAnnotatedNodesAndTokens(annotation).SingleOrDefault();
+                    var node = nodeOrToken.IsNode
+                        ? nodeOrToken.AsNode()
+                        : nodeOrToken.AsToken().Parent;
+
+                    if (node is null)
+                    {
+                        continue;
```

Comment:
hrmm... i guess that's what you're doing here.  so i think that's sound. but doc that this maybe indicates a bug, and give a Debug.Assert so we can try to track down.

Reply:
It's not a bug in this case. It's possible the symbol isn't in this document. Grouping document and symbol together would help with clarity here

Reply:
Added grouping and changed to being an assert